packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/services/RepresentationDataUpdateService.java

```diff
@@ -47,8 +47,10 @@ public IResult<Void> updateContent(UUID id, String content) {
         var optionalRepresentationData = this.representationDataRepository.findById(id);
         if (optionalRepresentationData.isPresent()) {
             var representationData = optionalRepresentationData.get();
-            representationData.updateContent(content);
-            this.representationDataRepository.save(representationData);
+            if (!representationData.getContent().equals(content)) {
+                representationData.updateContent(content);
+                this.representationDataRepository.save(representationData);
+            }
```

Comment:
This should be in RepresentationData#updateContent

Reply:
Done.